### DISCUSSION TITLE
Cannot get the Arducam Hawk-Eye (64MP) to work properly on a RPI 4 with Bullseye

### DISCUSSION THREAD

**Discussion by WebfinityDesign** [photobooth-app/photobooth-app#306] [Howtos and camera specific topics]
When starting the app I do have a preview but when I take a picture I get "streaming exception: device raised exception". After that I lose the preview.
Running on a RPI 4 with Bullseye. Tried with the Bullseye drivers and with the Arducam drivers.

Log:
`Traceback (most recent call last):
  File "/home/theo/.local/lib/python3.9/site-packages/photobooth/services/backends/abstractbackend.py", line 343, in wait_for_lores_image
    return self._wait_for_lores_image()  # blocks 0.5s usually. 10 retries default wait time=5s
  File "/home/theo/.local/lib/python3.9/site-packages/photobooth/services/backends/picamera2.py", line 246, in _wait_for_lores_image
    raise TimeoutError("timeout receiving frames")
TimeoutError: timeout receiving frames

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/theo/.local/lib/python3.9/site-packages/photobooth/services/aquisitionservice.py", line 194, in _get_stream_from_backend
    output_jpeg_bytes = backend_to_stream_from.wait_for_lores_image()
  File "/home/theo/.local/lib/python3.9/site-packages/photobooth/services/backends/abstractbackend.py", line 353, in wait_for_lores_image
    raise RuntimeError("device raised exception") from exc
RuntimeError: device raised exception
2024-06-21 14:11:15,837 [   ERROR] streaming exception: device raised exception (aquisitionservice.py:201)`

**Comment by mgineer85** [comment]
Could you please post all the logs?
Did arducam still not manage to update the driver to bookworm?
Picamera2 does not receive any updates to bullseye any more, so maybe you encounter an issue due to old dependencies.

**Comment by WebfinityDesign** [comment]
I started with Bookworm but had some trouble getting my touchscreen to work, so I switched to Bullseye. Ordered another touchscreen, so if Bookworm is better, I will try to switch to bookworm again.
Here is the log:
[Log 21-6.txt](https://github.com/user-attachments/files/15928706/Log.21-6.txt)

**Comment by WebfinityDesign** [comment]
If found this. Could number "7. Cannot Allocate Memory on Arducam 64MP Hawkeye Autofocus Camera Module"
have something to do with the problem?

UPDATE: I used this link [https://docs.arducam.com/Raspberry-Pi-Camera/Native-camera/64MP-Hawkeye/](url) and now it's working on Bookworm.

I think especially this part is important:

`sudo nano /boot/firmware/config.txt 

#Find the line: [all], add the following item under it:

dtoverlay=arducam-64mp

#Save and reboot.
`

**Reply by mgineer85** [reply]
Glad it’s working.
On long term I recommend update to bookworm if arducam is compatible because it is the only system with up to date picamera.

**Comment by WebfinityDesign** [comment]
I am no coder, so I can't help you that way, but if you want to have something tested on the Arducam 64MP just let me know, I'm glad to help
